src/core/raster/qgsrasterlayertemporalproperties.h

```diff
@@ -0,0 +1,122 @@
+/***************************************************************************
+                         qgsrasterlayertemporalproperties.h
+                         ---------------
+    begin                : February 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef QGSRASTERLAYERTEMPORALPROPERTIES_H
+#define QGSRASTERLAYERTEMPORALPROPERTIES_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgsrange.h"
+#include "qgsmaplayertemporalproperties.h"
+
+/**
+ * \class QgsRasterLayerTemporalProperties
+ * \ingroup core
+ * Implementation of map layer temporal properties for raster layers.
+ *
+ * \since QGIS 3.14
+ */
+
+class CORE_EXPORT QgsRasterLayerTemporalProperties : public QgsMapLayerTemporalProperties
+{
+  public:
+
+    /**
+     * Constructor for QgsRasterLayerTemporalProperties.
+     */
+    QgsRasterLayerTemporalProperties( bool enabled = false );
+
+    ~QgsRasterLayerTemporalProperties();
+
+    enum TemporalMode
+    {
+      ModeFixedTemporalRange,
```

Comment:
You'll need to add docstrings for each of these too:

```suggestion
      ModeFixedTemporalRange, //!< Explanation of the value
```